File: android/app/src/main/java/com/stalkerhek/app/engine/EngineBridge.java
package com.stalkerhek.app.engine;

/**
 * JNI bridge to the native stalkerhek Go engine compiled as a shared library.
 * The .so is bundled in jniLibs/<abi>/libstalkerhek_engine.so and loaded via
 * System.loadLibrary at class init time.
 */
public class EngineBridge {
    static { System.loadLibrary("stalkerhek_engine"); }

    /** Initialize engine with data directory. Returns JSON: {"ok":"true","profiles_loaded":N} */
    public static native String nativeInit(String dataDir);

    /** Start a profile. Accepts ProfileConfig JSON, returns ProfileStatus JSON. */
    public static native String nativeStartProfile(String profileJson);

    /** Stop a profile by ID. Returns {"ok":"true"} or {"ok":"false"}. */
    public static native String nativeStopProfile(int profileId);

    /** Get channels for a profile by media type ("itv","vod","series"). Returns JSON array. */
    public static native String nativeGetChannels(int profileId, String mediaType);

    /** Get all profiles. Returns JSON array of ProfileConfig. */
    public static native String nativeGetProfiles();

    /** Get profile status by ID. Returns ProfileStatus JSON. */
    public static native String nativeGetProfileStatus(int profileId);

    /** Create or update a profile. Returns ProfileConfig JSON. */
    public static native String nativeCreateProfile(String profileJson);

    /** Delete a profile by ID. */
    public static native String nativeDeleteProfile(int profileId);

    /** Gracefully shutdown engine. */
    public static native String nativeShutdown();
}

File: android/app/src/main/java/com/stalkerhek/app/engine/EngineBridge.kt
package com.stalkerhek.app.engine

object EngineBridge {
    init {
        System.loadLibrary("stalkerhek_engine")
    }

    /** Initialize the engine with Android internal storage path. Returns JSON status. */
    external fun nativeInit(dataDir: String): String

    /** Start a profile. Accepts ProfileConfig JSON, returns start result JSON. */
    external fun nativeStartProfile(profileJson: String): String

    /** Stop a profile by id. Returns {"ok":true/false}. */
    external fun nativeStopProfile(profileId: Int): String

    /** Get channels for a profile. type = "itv", "vod", or "series". Returns JSON array. */
    external fun nativeGetChannels(profileId: Int, mediaType: String): String

    /** Get all profiles. Returns JSON array of ProfileConfig. */
    external fun nativeGetProfiles(): String

    /** Get profile status by id. Returns ProfileStatus JSON. */
    external fun nativeGetProfileStatus(profileId: Int): String

    /** Create or update a profile. Returns ProfileConfig JSON. */
    external fun nativeCreateProfile(profileJson: String): String

    /** Delete a profile by id. Returns {"ok":true/false}. */
    external fun nativeDeleteProfile(profileId: Int): String

    /** Gracefully shut down the engine (stop servers, release resources). */
    external fun nativeShutdown(): String
}
